Gemfile

```diff
@@ -31,6 +31,7 @@ gem 'jbuilder', '~> 2.5'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 gem 'devise'
+gem 'rails-controller-testing'
```

Comment:
You shouldn't use this gem anymore. You don't need it for request specs. 

It is a gem to restore some functionality that was removed from rails.
Basically, rails has removed controller specs, which used to reach into the controller class and check the value of instance variables (`with assigns`) - very weird hence its removal. This gem restores that weird ability 👎 .

Reply:
I installed the gem as I get this error without it - 
```
Failure/Error: expect(response).to render_template(:new)

NoMethodError:
       assert_template has been extracted to a gem. To continue using it,
               add `gem 'rails-controller-testing'` to your Gemfile.
```

Reply:
I have just removed the assert_template related assertions from my tests :)

Reply:
`assert_template` was another weird rails method that allowed you to check what template the controller renders.

The reason this (and `assigns`) are weird is that they are checking not the return value of the method or the side effects of the method, but instead are checking the bits of logic *inside* the method, which normal unit tests and feature testing never do.